src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```diff
@@ -129,99 +126,107 @@ public void removeEventListener(EventListener listener) {
     }
 
     /**
-     * Helper method that returns the current number of subscribed
-     * listeners.
+     * Helper method that returns the current number of subscribed listeners.
+     *
      * @return The current number listeners.
      */
     public int getListenerCount() {
-        return listeners.size();
+        int size = 0;
+        synchronized (listeners) {
+            size = listeners.size();
+        }
+        return size;
+    }
+
+    /**
+     * Helper method to check if the WebSocket Session exists and is open.
+     *
+     * @return A flag indicating the {@link EventStream}
+     * WebSocket {@link Session} state.
+     */
+    public boolean isEventStreamClosed() {
```

Comment:
I'm fixing the code with this.
Sure, I will take care about brackets; I'm checking if any other statement needs them too.